src/RequestBuilder/RecommendationRequestBuilder.php

```diff
@@ -72,6 +74,13 @@ private function assertInteractionUserId(): void
 
         $interactionUserId = $this->interactionCommand->getUserId();
 
+        // allow (B, A -> B, B)
```

Comment:
Maybe move this after the following condition, so that cases for "not null user-merge" are next to each other?

Reply:
moved